quesma/queryparser/pancake_model.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package queryparser
+
+import (
+	"quesma/logger"
+	"quesma/model"
+)
+
+type pancakeAggregationTopLevel struct {
+	children    []*pancakeAggregationLevel
+	whereClause model.Expr
+}
+
+type pancakeAggregationLevel struct {
+	name            string
+	queryType       model.QueryType
+	selectedColumns []model.Expr
+
+	// only for bucket aggregations
+	children []*pancakeAggregationLevel
+	orderBy  []model.OrderByExpr
+	limit    int // 0 if none, only for bucket aggregation
+	isKeyed  bool
+
+	metadata    model.JsonMap
+	whereClause model.Expr
+}
+
+type pancakeFillingMetricAggregation struct {
+	name            string
+	queryType       model.QueryType // it has to be metric aggregation
+	selectedColumns []model.Expr
+
+	metadata model.JsonMap
+}
+
+type pancakeLayerBucketAggregation struct {
+	name            string
```

Comment:
`pancakeFillingMetricAggregation` contains all fields of pancakeFillingMetricAggregation, so the latter can be embedded

```suggestion
	pancakeFillingMetricAggregation
```